pkg/sbom/spdx/marshal.go

```diff
@@ -333,9 +333,17 @@ func (m *Marshaler) pkgToSpdxPackage(t ftypes.TargetType, pkgDownloadLocation st
 		pkgSrcInfo = fmt.Sprintf("%s: %s %s", SourcePackagePrefix, pkg.SrcName, utils.FormatSrcVersion(pkg))
 	}
 
-	packageURL, err := purl.NewPackageURL(t, metadata, pkg)
-	if err != nil {
-		return spdx.Package{}, xerrors.Errorf("failed to parse purl (%s): %w", pkg.Name, err)
+	var packageURL *purl.PackageURL
+	if !pkg.Identifier.Empty() && pkg.Identifier.PURL != "" {
+		packageURL, err = purl.FromString(pkg.Identifier.PURL)
+		if err != nil {
+			return spdx.Package{}, xerrors.Errorf("failed to create package purl: %w", err)
+		}
+	} else {
+		packageURL, err = purl.NewPackageURL(t, metadata, pkg)
+		if err != nil {
+			return spdx.Package{}, xerrors.Errorf("failed to create package purl: %w", err)
+		}
```

Comment:
I don't think this makes sense.
We already tried to create purl in `analyzer`.

For SDPX  we can remove this and leave `packageURL == nil`. In this case we simply don't include purl to SPDX package:
https://github.com/aquasecurity/trivy/blob/c866f1c4e99ba0026fe9f632fbe97fc5d89165d1/pkg/sbom/spdx/marshal_test.go#L894-L903

Reply:
addressed at https://github.com/aquasecurity/trivy/pull/5439/commits/cdd281e1e8fc988cf747b8a94e734e548c4ee310